src/GuzzleHttp/Middleware/StopwatchMiddleware.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the CsaGuzzleBundle package
+ *
+ * (c) Charles Sarrazin <charles@sarraz.in>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code
+ */
+
+namespace Csa\Bundle\GuzzleBundle\GuzzleHttp\Middleware;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+/**
+ * Stopwatch Middleware.
+ *
+ * @author Charles Sarrazin <charles@sarraz.in>
+ */
+class StopwatchMiddleware
+{
+    private $stopwatch;
+    private $increments = [];
+
+    public function __construct(Stopwatch $stopwatch)
+    {
+        $this->stopwatch = $stopwatch;
+    }
+
+    public function __invoke(callable $handler)
+    {
+        return function (RequestInterface $request, array $options) use ($handler) {
+            $key = sprintf('%s %s', $request->getMethod(), (string) $request->getUri());
+
+            if (!isset($this->increments[$key])) {
+                $this->increments[$key] = 0;
+            }
+
+            ++$this->increments[$key];
+
+            $key .= ' ('.$this->increments[$key].')';
+
+            if (!$this->stopwatch->isStarted($key)) {
```

Comment:
IMO this condition should be removed. The section must be unique (which this PR fixes) so a section with the same name is not supposed to be started already.